kotest-extensions/src/jvmMain/kotlin/io/kotest/extensions/system/SystemEnvironmentExtensions.kt

```diff
@@ -183,6 +184,30 @@ class SystemEnvironmentTestListener(environment: Map<String, String>, mode: Over
    override suspend fun afterTest(testCase: TestCase, result: TestResult) {
       resetSystemEnvironment()
    }
+
+   override suspend fun beforeContainer(testCase: TestCase) {
+      if (testCase.type == TestType.Container) changeSystemEnvironment()
+   }
+
+   override suspend fun afterContainer(testCase: TestCase, result: TestResult) {
+      if (testCase.type == TestType.Container) resetSystemEnvironment()
+   }
+
+   override suspend fun beforeEach(testCase: TestCase) {
+      if (testCase.type == TestType.Test) changeSystemEnvironment()
+   }
+
+   override suspend fun afterEach(testCase: TestCase, result: TestResult) {
+      if (testCase.type == TestType.Test) resetSystemEnvironment()
```

Comment:
We probably only need to call changeSystemEnvironment /  resetSystemEnvironment on one of the beforeAny / beforeTest cases as that will cover it for every test.